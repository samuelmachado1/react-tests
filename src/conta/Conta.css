.Conta-header {
    color: #6633FF;
}

.Saldo-valor {
    color: #426fa2;
}