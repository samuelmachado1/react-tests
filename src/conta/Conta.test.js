import React from "react";
import { render } from "@testing-library/react";
import Conta from "./Conta";

describe("Componente de saldo da conta", () => {
  it("O Snapshot do component deve permanecer sempre o mesmo", () => {
    const { container } = render(<Conta saldo="100" />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
