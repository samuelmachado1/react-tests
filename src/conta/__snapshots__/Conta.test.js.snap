// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente de saldo da conta O Snapshot do component deve permanecer sempre o mesmo 1`] = `
<div
  class="Conta-header"
>
  <h2>
    Conta
  </h2>
  <p>
    Saldo: 
    <span
      class="Saldo-valor"
      data-testid="saldo-conta"
    >
      R$ 100
    </span>
  </p>
  <form>
    <div>
      <label>
        Depósito
        <input
          data-testid="transacao"
          name="transacao"
          type="radio"
          value="deposito"
        />
      </label>
    </div>
    <div>
      <label>
        Saque
        <input
          data-testid="transacao"
          name="transacao"
          type="radio"
          value="saque"
        />
      </label>
    </div>
    <label>
      Valor:
    </label>
    <input
      data-testid="valor"
      name="valor"
      type="text"
      value="0"
    />
    <div>
      <button
        type="submit"
      >
        Realizar operação
      </button>
    </div>
  </form>
</div>
`;

exports[`Componente de saldo da conta O Snapshot do componente deve permanecer sempre o mesmo 1`] = `
<div
  class="Conta-header"
>
  <h2>
    Conta
  </h2>
  <p>
    Saldo: 
    <span
      class="Saldo-valor"
      data-testid="saldo-conta"
    >
      R$ 100
    </span>
  </p>
  <form>
    <div>
      <label>
        Depósito
        <input
          data-testid="transacao"
          name="transacao"
          type="radio"
          value="deposito"
        />
      </label>
    </div>
    <div>
      <label>
        Saque
        <input
          data-testid="transacao"
          name="transacao"
          type="radio"
          value="saque"
        />
      </label>
    </div>
    <label>
      Valor:
    </label>
    <input
      data-testid="valor"
      name="valor"
      type="text"
      value="0"
    />
    <div>
      <button
        type="submit"
      >
        Realizar operação
      </button>
    </div>
  </form>
</div>
`;
